hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieCopyOnWriteTableInputFormat.java

```diff
@@ -65,12 +71,32 @@
  *   <li>Incremental mode: reading table's state as of particular timestamp (or instant, in Hudi's terms)</li>
  *   <li>External mode: reading non-Hudi partitions</li>
  * </ul>
+ *
+ * NOTE: This class is invariant of the underlying file-format of the files being read
  */
-public abstract class HoodieFileInputFormatBase extends FileInputFormat<NullWritable, ArrayWritable>
+public class HoodieCopyOnWriteTableInputFormat extends FileInputFormat<NullWritable, ArrayWritable>
```

Comment:
Can you elaborate what you see confusing in there? We already have such splitting in Spark for ex (`MergeOnReadSnapshotRelation`, etc)

I actually think it's much more cleaner connecting w/ MOR/COW dichotomy rather previous one of Realtime/non-Realtime

Reply:
I agree that `Realtime` naming can be removed.

I come from the angle where the InputFormat classes can be named according to the Hudi file layouts, i.e., file groups with file slices containing a base file and a set of log files.  "CopyOnWrite" and "MergeOnRead" naming is one layer above for write/read logic.  They are cleaner than the previous naming.  I'm thinking that "BaseFile" and "BaseAndLogFile" prefixes may be a better fit here.

Since the changes are fundamental, I prefer that the naming should be finalized in 0.11.0 and won't be changed for some time.  So consensus is needed here.

Reply:
I see your point. However, the affiliation with COW/MOR is not about the semantic here, but rather reference to the _table's type_ which is pivotal here -- InputFormats are about reading Hudi's tables and it seems odd that we're not aliasing the table in any way with the abstraction that is supposed to read this very table.

Happy to jump on a call and jam more to find the convergence here.

Reply:
Actually table types are ONLY about how data is stored, not how they are queried. MOR's read optimized queries are now served via "CopyOnWrite" ? https://hudi.apache.org/docs/table_types#table-and-query-types  So not sure I agree with the reasoning above.

But incremental query is now overloaded onto the input format. So this renaming is ok for now

Reply:
Yeah, i've crossed the same path recently realizing that this dichotomy doesn't line up well with the reading path, and i think the crux of the problem is that COW is purely write-side semantic therefore when we say COW on the read-path that doesn't really make sense. 

I'm touching up sibling hierarchy on Spark side, and will think about better terminology there and afterwards we can carry it over here as well